File: 298-binary-tree-longest-consecutive-sequence/298-binary-tree-longest-consecutive-sequence.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def longestConsecutive(self, root: Optional[TreeNode]) -> int:
        self.longest = 0
        def dfs(root, parent, curr_value):
            if root:
                if parent:
                    if root.val == parent.val + 1:
                        curr_value += 1
                    else:
                        curr_value = 1
            self.longest = max(self.longest, curr_value)
            if root.left:
                dfs(root.left, root, curr_value)
            if root.right:
                dfs(root.right, root, curr_value)
        
        dfs(root, None, 1)
        return self.longest
                            
        
        

File: 298-binary-tree-longest-consecutive-sequence/298-binary-tree-longest-consecutive-sequence.go
var ans int

func longestConsecutive(root *TreeNode) int {
    
    ans= 0
    
    if root==nil{
        return 0
    }
    
    helper(root, 1) 
    return ans
    
}

func helper(root *TreeNode, count int) {
    
    ans = max(ans, count)
    
    
    if root.Left!=nil && (root.Left.Val == root.Val + 1) {
        helper(root.Left, count + 1)
    }else{
        if root.Left!=nil{
            helper(root.Left, 1)
        }
    }
    
    if root.Right!=nil && (root.Right.Val== root.Val + 1){
        helper(root.Right, count + 1)
    }else{
        if root.Right!=nil{
            helper(root.Right, 1)
        }
    }
    
}

func max(i, j int) int{
    if i> j{
        return i
    }
    
    return j
}